src/main/ruby/core/enumerable.rb

```diff
@@ -112,6 +112,7 @@ def count(item = undefined)
     seq = 0
     if !undefined.equal?(item)
       each { |o| seq += 1 if item == o }
+      warn 'warning: given block not used' if block_given?
```

Comment:
I think it should also warn as soon as possible, and in particular before calling `each`.